### PR TITLE
Add support for numpy 2.0

### PR DESCRIPTION
**Pull request by allevitan**
Makes some changes to support numpy 2.0, and fixes a few old issues where pytorch tensors converted to numpy arrays where they didn't need to be.

### CODE REVIEW

File: src/cdtools/tools/data/data.py [hunk 1]
```diff
@@ -134,18 +134,18 @@ def get_sample_info(cxi_file):
             metadata[attr] = np.float32(s1[attr][()])
 
     if 'unit_cell' in s1:
-        metadata['unit_cell'] = np.array(s1['unit_cell']).astype(np.float32)
+        metadata['unit_cell'] = s1['unit_cell'][()].astype(np.float32)
```
Comment (clemisch) [inline src/cdtools/tools/data/data.py:137]: What is the `[()]` for? 

Unlike `array[:]`, it does not seem to create a copy of the array, if that is the goal here.

Comment (clemisch) [reply src/cdtools/tools/data/data.py:137]: Sorry, I just checked and `array[:]` apparently also doesn't create a copy (anymore??). So I guess you can disregard my comment, although I am still interested in the purpose of `[()]`.

Comment (allevitan) [reply src/cdtools/tools/data/data.py:137]: IIRC, s1['unit_cell'] returns an h5py dataset, this converts it to a numpy array. See Note 2 from [this stackexchange question](https://stackoverflow.com/a/46733277), also the [release notes for h5py 2.1](https://docs.h5py.org/en/stable/whatsnew/2.1.html#dataset-value-property-is-now-deprecated)

**Review by clemisch — APPROVED** [review]
lgtm

**Pull request merged**